src/rdkafka_int.h

```diff
@@ -339,6 +337,42 @@ struct rd_kafka_s {
                 /**< Lock for sparse_connect_random */
                 mtx_t         sparse_connect_lock;
         } rk_suppress;
+
+        /**
+         * SASL/OAUTHBEARER token value, token metadata, and extensions
+         */
+        struct {
+                /**< The lock protecting the below information */
+                rwlock_t refresh_lock;
```

Comment:
I don't think we need a dedicated lock for this, instead use rd_kafka_wrlock() and rd_kafka_rdlock().

Reply:
👍